### ISSUE TITLE
`uninlined_format_args` false negatives

### ISSUE THREAD

**Issue by mkroening** [rust-lang/rust-clippy#16411]
### Summary

This issue is three false negatives in a trench coat.
I can split them up into separate issues, though, if you prefer that. :)

1. `uninlined_format_args` does not trigger on `std::panic!` when using a format string without text (`"{}"`) on Edition 2024. [Edition 2021's panic macro consistency](https://doc.rust-lang.org/edition-guide/rust-2021/panic-macro-consistency.html) should not matter.
2. `uninlined_format_args` does not trigger on `core::panic!` even when using a format string with text (`"foo{}"`) which works with `std::panic!`.
3. `uninlined_format_args` does not trigger on macros that have the `#[clippy::format_args]` attribute when they use the format args nestedly without binding the inner args first.

Thanks for your work! :)

### Lint Name

uninlined_format_args

### Reproducer

I tried this code:

```rust
//! `foo.rs`

pub fn a(a: i32) {
    // Warns correctly when using a non-empty format string instead.
    // For example, `foo{}`, as done in `b`.
    std::panic!("{}", a);
}

pub fn b(b: i32) {
    // Warns correctly, when using `std::panic!` instead.
    core::panic!("foo{}", b);
}

#[clippy::format_args]
macro_rules! macro_c {
    ($($arg:tt)*) => {{
        // Warns correctly, when binding the value first instead:
        // match ::core::format_args!($($arg)*) {
        //     args => ::core::format_args!("{}\n", args),
        // }
        ::core::format_args!("{}\n", ::core::format_args!($($arg)*));
    }};
}

pub fn c(c: i32) {
    macro_c!("{}", c);
}
```

I ran Clippy like this:

```bash
clippy-driver +nightly --crate-type lib --edition 2024 -W clippy::uninlined_format_args foo.rs
```

I expected to see this happen:

An `uninlined_format_args` warning should be emitted for every one of the three functions.

Instead, this happened:

No warning was emitted for any of the functions.

### Version

```text
rustc 1.92.0 (ded5c06cf 2025-12-08)
binary: rustc
commit-hash: ded5c06cf21d2b93bffd5d884aa6e96934ee4234
commit-date: 2025-12-08
host: aarch64-apple-darwin
release: 1.92.0
LLVM version: 21.1.3
```

**Comment by nyurik** [comment]
fixed in #16597
